src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -351,6 +352,58 @@ public static void Parse(string value, NumberStyles style, IFormatProvider provi
             }
         }
 
+        internal static string SplitPairs(string input)
+        {
+            string[] splitPairs = input.Split('-');
+            string ret = "";
+            foreach (var pair in splitPairs)
+            {
+                string reversedPair = Reverse(pair);
+                ret += reversedPair;
+            }
+
+            return ret;
+        }
+
+        internal static string Reverse(string s)
+        {
+            char[] charArray = s.ToCharArray();
+            Array.Reverse(charArray);
+            return new string(charArray);
+        }
+
+        [Fact]
```

Comment:
I will consider this on my next pass through here. I actually tried putting the data from ibm-fpgen.txt into a static class to call with `ClassData` like you suggest, but VS turned out to be so sluggish that it was impossible to work on that file.

Reply:
Ha - I bet. You could also just have a class that reads the data from the text file and loads it into a well-formed structure though.

Reply:
Alright, I took a pass at this again and concluded that the sluggishness that VS exhibits is impossible to work with. @jeffhandley, do you mind signing off on the PR so I can merge when CI turns green?

Reply:
You could do
```
[Theory]
[MemberData(nameof(ParsePattern_TestData))]
```

with
```
public static IEnumerable<object[]> ParsePattern_TestData()
{
    string path = Directory.GetCurrentDirectory();
    using (FileStream file = new FileStream(Path.Combine(path, "NumericsTestData", "ibm-fpgen.txt"), FileMode.Open))
    {
        // logic
        yield return doubleBytes;
    }
}
```

Reply:
Yeah, I too was trying to suggest something like that where the data remains in a text file (not intended to be opened in Visual Studio), but that was loaded into a class/member instead of parsed as part of the test itself. It's OK with me to have that done in a follow-up PR though.